### PR TITLE
Move appnet client to ecs-agent

### PR DESCRIPTION
**Pull request by RichaGangwar**
### Summary
Move appnet client to ecs-agent

### Implementation details
1. Moves the folder agent/api/appnet to ecs-agent/api
2. Update AppnetClient's GetStats() and DrainInboundConnections() to accept required arguments instead of ServiceConnect runtime config. This is done to avoid the cyclic dependency between ecs-agent and agent.
3. Update all the corresponding usages of GetStats() and DrainInboundConnections() to reflect the changes.
4. Update the api_mocks.go by generating new mocks for api.
 
### Testing

Unit tested the changes.
New tests cover the changes: No

### Description for the changelog

Move appnet client to ecs-agent

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/engine/docker_task_engine.go [hunk 1]
```diff
@@ -2065,7 +2065,9 @@ func (engine *DockerTaskEngine) stopContainer(task *apitask.Task, container *api
 	// Before attempting to stop any container, send drain signal for Appnet Agent to start draining connections
 	// (if not already in progress).
 	if task.IsServiceConnectEnabled() && !task.IsServiceConnectConnectionDraining() {
-		if err := engine.appnetClient.DrainInboundConnections(task.GetServiceConnectRuntimeConfig()); err != nil {
+		adminSocketPath := task.GetServiceConnectRuntimeConfig().AdminSocketPath
+		drainRequest := task.GetServiceConnectRuntimeConfig().DrainRequest
```
Comment (Realmonia) [inline agent/engine/docker_task_engine.go:2069]: Can you combine these two statement to only make one call?

File: agent/go.mod
Comment (Realmonia) [inline agent/go.mod:1]: Are the version changes necessary here?

File: ecs-agent/go.mod
Comment (Realmonia) [inline ecs-agent/go.mod:1]: Are the version changes necessary here?

File: agent/engine/docker_task_engine.go (hunk 1 above)
Comment (RichaGangwar) [reply agent/engine/docker_task_engine.go:2069]: Sure.

File: agent/go.mod
Comment (RichaGangwar) [reply agent/go.mod:1]: So, I did go mod tidy and go mod vendor, and these were the changes done by these commands. Should I revert these changes then?

**Review by Realmonia — APPROVED** [review]

**Review by fierlion — APPROVED** [review]

**Pull request merged**